internal/rukpak/source/containers_image.go

```diff
@@ -174,31 +169,60 @@ func (i *ContainersImageRegistry) unpackPath(bundleName string, digest digest.Di
 	return filepath.Join(i.bundlePath(bundleName), digest.String())
 }
 
-func resolveCanonicalRef(ctx context.Context, imgRef reference.Named, imageCtx *types.SystemContext) (reference.Canonical, error) {
+func resolveReferences(ctx context.Context, ref string, sourceContext *types.SystemContext) (reference.Named, reference.Canonical, bool, error) {
+	imgRef, err := reference.ParseNamed(ref)
+	if err != nil {
+		return nil, nil, false, reconcile.TerminalError(fmt.Errorf("error parsing image reference %q: %w", ref, err))
+	}
+
+	canonicalRef, isCanonical, err := resolveCanonicalRef(ctx, imgRef, sourceContext)
+	if err != nil {
+		return nil, nil, false, fmt.Errorf("error resolving canonical reference: %w", err)
+	}
+	return imgRef, canonicalRef, isCanonical, nil
+}
+
+func resolveCanonicalRef(ctx context.Context, imgRef reference.Named, imageCtx *types.SystemContext) (reference.Canonical, bool, error) {
```

Comment:
It's not clear what the `bool` return is for. There's no explanation (I assume it's `isCanonical`?)

Reply:
Yeah, its `isCanonical`. One way to tell is by looking at the call site in line 178.

But, this one is a bit weird because operator-controller doesn't actually use this variable. Keep moving up the stack and you'll see we discard its value on line 49.

As best I can, I'm trying to keep the function signatures the same between catalogd and operator-controller to facilitate merging later.